### PR TITLE
avoid generating unused warnings in Scala codegen

### PR DESCRIPTION
**Pull request by S11001001**
From #6907, these remove all `-Ywarn-unused` warnings in Scala codegen's output, with one exception, explained in 839fff38dfff5c4404f28205644b198d7421b145 .

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/TemplateCompanion.scala [hunk 1]
```diff
@@ -81,6 +81,7 @@ abstract class TemplateCompanion[T](implicit isTemplate: T <~< Template[T])
   }
 
   protected final def ` exercise`[ExOn, Out](
+      actor: Primitive.Party,
```
Comment (cocreature) [inline language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/TemplateCompanion.scala:84]: This is not part of the public API right? so this is not a breaking change

File: language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala [hunk 2]
```diff
@@ -30,7 +30,7 @@ object PackageIDsGen {
     val packageIdsSrc: Tree =
       q"""
         package ${Util.packageNameToRefTree(util.packageName)} {
-          private object `Package IDs` {
+          private[${Util.packageNameTailToRefTree(util.packageName)}] object `Package IDs` {
```
Comment (cocreature) [inline language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala:33]: maybe worth adding a comment that this is just to silence the warning?

File: language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/TemplateCompanion.scala (hunk 1 above)
Comment (S11001001) [reply language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/TemplateCompanion.scala:84]: Right.

File: language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala (hunk 2 above)
Comment (S11001001) [reply language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala:33]: Honestly this is just more explicit about what the scope is, the unscoped version is kind of a weird corner case in Scala that I still think shouldn't work at all, that we were using just to make the codegen slightly simpler.

Comment (cocreature) [reply language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala:33]: sounds reasonable, I guess a comment might still save someone from trying to simplify it again but I don’t feel strongly about this.

Comment (S11001001) [reply language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala:33]: I wish the best of fortune to anyone who would care to simplify any one of dozens of bits of apparent complexity in the codegen and shall soon be acquainted with the blunt force of the ensuing codegen-sample-app compiler errors. 🙂

Comment (leo-da) [reply language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala:33]: I don't know... we don't document every package private modifier :)
Why should it be different for the generated code.

**Pull request merged**